File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/http/HttpConnection.php
<?php 
namespace io\vertx\php\core\http;
use io\vertx\php\core\buffer\Buffer;
use io\vertx\php\core\net\SocketAddress;
class HttpConnection
{
    
    private function __construct()
    {

    }
    /**
     * ()
     * @return Void
     * methodSize : 1
     */
    public function close(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function closeHandler(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function exceptionHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return int
     * methodSize : 1
     */
    public function getWindowSize(/*0*/)
    {

    }
    /**
     * (num errorCode)
     * @return HttpConnection
     * (num errorCode, int lastStreamId)
     * @return HttpConnection
     * (num errorCode, int lastStreamId, Buffer debugData)
     * @return HttpConnection
     * methodSize : 3
     */
    public function goAway(/*3*/$arg0, $arg1, $arg2)
    {

    }
    /**
     * (callable handler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function goAwayHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return string
     * methodSize : 1
     */
    public function indicatedServerName(/*0*/)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isSsl(/*0*/)
    {

    }
    /**
     * ()
     * @return SocketAddress
     * methodSize : 1
     */
    public function localAddress(/*0*/)
    {

    }
    /**
     * (Buffer data, callable pongHandler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function ping(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (callable handler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function pingHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return SocketAddress
     * methodSize : 1
     */
    public function remoteAddress(/*0*/)
    {

    }
    /**
     * ()
     * @return Http2Settings
     * methodSize : 1
     */
    public function remoteSettings(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function remoteSettingsHandler(/*1*/$arg0)
    {

    }
    /**
     * (int windowSize)
     * @return HttpConnection
     * methodSize : 1
     */
    public function setWindowSize(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return Http2Settings
     * methodSize : 1
     */
    public function settings(/*0*/)
    {

    }
    /**
     * ()
     * @return HttpConnection
     * (num timeoutMs)
     * @return HttpConnection
     * methodSize : 2
     */
    public function shutdown(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return HttpConnection
     * methodSize : 1
     */
    public function shutdownHandler(/*1*/$arg0)
    {

    }
    /**
     * (Http2Settings settings)
     * @return HttpConnection
     * (Http2Settings settings, callable completionHandler)
     * @return HttpConnection
     * methodSize : 2
     */
    public function updateSettings(/*2*/$arg0, $arg1)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/http/HttpConnection.java
package io.vertx.php.core.http;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.buffer.Buffer;
import io.vertx.php.core.net.SocketAddress;
@Name("HttpConnection")
@Namespace("io\\vertx\\php\\core\\http")
public interface HttpConnection<S extends io.vertx.core.http.HttpConnection> extends IWrapper<S>{
    
    @Signature
    default Memory getWindowSize(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().getWindowSize()
    }

    @Signature
    default Memory setWindowSize(Environment __env__, Memory... args) {
        this.getWrappedObject().setWindowSize(windowSize)
    }

    @Signature
    default Memory goAway(Environment __env__, Memory... args) {
        this.getWrappedObject().goAway(errorCode)
    }

    @Signature
    default Memory goAwayHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().goAwayHandler(handler)
    }

    @Signature
    default Memory shutdownHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().shutdownHandler(handler)
    }
    
    @Signature
    default Memory shutdown(Environment __env__, Memory... args) {
        this.getWrappedObject().shutdown()
    }

    @Signature
    default Memory shutdown(Environment __env__, Memory... args) {
        this.getWrappedObject().shutdown()
    }

    @Signature
    default Memory closeHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().closeHandler(handler)
    }
    
    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }
    
    @Signature
    default Memory settings(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().settings()
    }

    @Signature
    default Memory updateSettings(Environment __env__, Memory... args) {
        this.getWrappedObject().updateSettings(settings)
    }
    
    @Signature
    default Memory remoteSettings(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().remoteSettings()
    }

    @Signature
    default Memory remoteSettingsHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().remoteSettingsHandler(handler)
    }

    @Signature
    default Memory ping(Environment __env__, Memory... args) {
        this.getWrappedObject().ping(data, pongHandler)
    }

    @Signature
    default Memory pingHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().pingHandler(handler)
    }

    @Signature
    default Memory exceptionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().exceptionHandler(handler)
    }
    
    @Signature
    default Memory remoteAddress(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, SocketAddressImpl::new, this.getWrappedObject().remoteAddress()
    }
    
    @Signature
    default Memory localAddress(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, SocketAddressImpl::new, this.getWrappedObject().localAddress()
    }
    
    @Signature
    default Memory isSsl(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isSsl()
    }
    
    @Signature
    default Memory indicatedServerName(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().indicatedServerName()
    }

}
